datapower-docker-compose-dp-mq/datapower-docker-compose-dp-mq.mkd

```diff
@@ -0,0 +1,151 @@
+curldriver_1  | response 1 received
+```
+In the message being returned by the `bob_soapui_1` container, you can see if the test passed by looking for the VALID status of the Assertion (fourth line from the bottom in the output above).
+
+Depending on the system on which you are deploying, the first message or two may fail to be processed as the environment may still be building. This is easily dectable in the console logs.
+
+When the fifth message has been processed you may exit the test with the docker command ctrl+c, which will exit and stop the containers. To rerun the test, run the docker-compose up command again.
+
+## Request Flow- Meet bob
+Here I'll outline the travels of the request message, `bob`, through the pipeline created in the docker-compose environment. 
+
+First the `bob_curldriver_1` container runs the following shell script.
+```
+ #!/bin/bash
+sleep 30
+
+for (( i=1; i <= 5; i++ ))
+  do
+   echo "sending message $i"
+     curl --silent --insecure --form "project=@/usr/local/bin/grow-soapui-project.xml" --form "suite=sendRequestGrow" http://soapui:3000 || echo curl error $?
+   echo "response $i received"
+  sleep 5
+  done
+```
+The script curls a SoapUI project to `bob_soapui_1`. The script is set to run 5 times with a five second delay between each. These parameters are adjustable to your needs/liking.
+
+`bob_soapui_1` accepts the curl request and processes the test defined in the SoapUI project suite. This is where Bob is created as an XML message and sent to `bob_datapower_1`.
+
+`bob_datapower_1` accepts Bob from `bob_soapui_1`. The processing policy retrieves various styles heets and schema from `bob_http_1` and transforms Bob. The policy itself is simple. As Bob passes through each rule in the policy, his number of nodes and elements increase. At the end, Bob is a message that reports what size he was in bytes at the end of the trip through DataPower. Bob is then routed to the `bob_mq_1` back end.
+
+Bob is placed on a queue in `bob_mq_1` and his travel could very well end here; however, to avoid the queue getting full on a long running test, I set up a dummy service on `bob_datapower_1` that clears the queue and throws the message away. 
+
+## Making Changes- You have the power
+The power to make changes in this sample is up to you. The construction of the environment lends itself to customization. The processing policy in the DataPower configuration can be replaced; the SoapUI project can be changed with a different test or additional requests could be added; the curl driver parameters can be adjusted to change the rate and duration of test messages; multiple MQ queue managers can be added. The list is endless and completely up to you.
+
+##Cleaning Up
+When you have completed your testing, you can remove the containers created during the exercise with the following command:
+```
+docker-compose -p bob rm
+```
+You can rerun the environment by issuing the docker-compose run command again.
```

Comment:
Some general things seem to be missing:

- [ ] Need some kind of closing or conclusion
- [ ] Need to point out where the different config is located and what it does
- [ ] Should link to the soapui stuff you used and their directions